framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3106,7 +3106,7 @@ public boolean validateTypeOf(Tree tree) {
             }
             break;
         default:
-            type = atypeFactory.getAnnotatedType(tree);
+        	type = atypeFactory.getDefaultedAnnotatedType(tree);
```

Comment:
Thanks for the test case!  I think the BaseTypeValidator validates all types, even if not explicitly written.  For example:
````
@Nullable T extends Number
````
The annotation on Number is defaulted to `@NonNull`, so the bounds of `T` are not valid.  The base type visitor issues a warning.  

Given this code, which might be in byte code.
```` 
@H1S2 Object getH1S2() {
   return new @H1S2 Object();
}
````
Should `o1 = getH1S2()` issue a warning, too?  

PR #663 is my preferred way of avoiding the error on `o2`.